Src/Base/AMReX_CArena.cpp

```diff
@@ -369,25 +382,13 @@ CArena::freeUnused_protected ()
                                      {
                                          it = m_freelist.erase(it);
                                          nbytes += a.second;
-                                         to_free.emplace_back(a.first, a.second);
+                                         deallocate_system(a.first,a.second);
```

Comment:
I think this could deadlock if an Elixir were to be used outside an MFIter

Reply:
In principle, yes. But we do not use multiple streams outside MFIter.

Reply:
Even with a single stream I suspect it is possible
```C++
{
    Elixir eli(...);
    ParallelFor(...);
}
... = carena->alloc(...); // triggers freeUnused_protected -> deallocate_system ->
// cudaFree -> deviceSync -> elixir free -> carena free -> needs carena mutex, deadlock
```